src/main/java/com/github/cypher/sdk/Client.java

```diff
@@ -1,13 +1,242 @@
 package com.github.cypher.sdk;
 
 import com.github.cypher.sdk.api.ApiLayer;
+import com.github.cypher.sdk.api.RestfulHTTPException;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.reflect.TypeToken;
+import com.sun.javafx.collections.ObservableMapWrapper;
+import javafx.collections.MapChangeListener;
+import javafx.collections.ObservableMap;
 
-// Poses as a chat instance, being the object returned from this package
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class takes an ApiLayer object and parses
+ * the data it returns into usable objects
+ */
 public class Client {
 
-    private ApiLayer api;
+	private ApiLayer api;
+	private String lastSyncMarker = null;
+
+	private static final String settingsNamespace = "com.github.cypher.settings";
```

Comment:
I don't think this belongs in the SDK as it should be independently redistributable.